src/features/web-compat.js

```diff
@@ -72,6 +72,29 @@ function permissionsFix (settings) {
     }
 }
 
+function messageHandlersFix () {
+    // Do nothing if `messageHandlers` is absent
+    if (!window.webkit?.messageHandlers) return
+
+    // handlers in any page-world user-scripts
+    const knownHandlers = ['trackerDetectedMessage', 'printHandler', 'specialPages']
+
+    window.webkit = {
+        messageHandlers: new Proxy(window.webkit.messageHandlers, {
+            get (target, messageName, receiver) {
+                if (knownHandlers.includes(String(messageName))) {
+                    return Reflect.get(target, messageName, receiver)
```

Comment:
It seems like there's three potential strategies here we could take (for the short term).

Return undefined, return a blank post message and return the real listener. Is it easy enough to make each three configurable based on name passed in from the config.

Reply:
@jonathanKingston yes I like the idea of more things configurable, but I'm not exactly sure what you mean - are you talking about when we return for the dynamic name part? 

From testing on that site (and their less-than-ideal js) we need to ensure the entire access path of `window.webkit.messageHandlers.<name>.postMessage` returns a function, otherwise it will continue to throw

Reply:
We can drop the third one I mentioned as you say just having window.webkit is less than ideal anyway that we should fix in itself.

Reply:
Strategies are what we return:

1. undefined
2. `{ postMessage () {...}}`
3. `Reflect.get(target, messageName, receiver)`

Then I was thinking the config would be something like:
```
{
    "webCompat": {
        "settings": {
            "messageHandlers" {
                "state": "enabled",
                "handlerStrategies": {
                    undefined: ['boop'],
                    reflect: ['trackerDetectedMessage', 'printHandler', 'specialPages'],
                    polyfil: ['*']
                }
            }
        }
    }
}
```